crates/revm/src/optimism.rs

```diff
@@ -83,19 +77,16 @@ impl L1BlockInfo {
     }
 
     /// Calculate the gas cost of a transaction based on L1 block data posted on L2
-    pub fn calculate_tx_l1_cost<SPEC: Spec>(&self, input: &Bytes, is_deposit: bool) -> U256 {
-        let rollup_data_gas_cost = self.data_gas::<SPEC>(input);
-
-        if is_deposit || rollup_data_gas_cost == U256::ZERO {
-            return U256::ZERO;
-        }
+    pub fn calculate_tx_l1_cost<SPEC: Spec>(&self, input: &Bytes) -> U256 {
+        // input must not be an deposit transaction
+        debug_assert!(!input.is_empty() && input[0] != 0x7E);
 
+        let rollup_data_gas_cost = self.data_gas::<SPEC>(input);
         rollup_data_gas_cost
             .saturating_add(self.l1_fee_overhead)
             .saturating_mul(self.l1_base_fee)
             .saturating_mul(self.l1_fee_scalar)
-            .checked_div(U256::from(1_000_000))
-            .unwrap_or_default()
+            / U256::from(1_000_000)
```

Comment:
Would prefer if we kept this as a checked div just in case.

Reply:
`checked_div` only checks if the divisor is `0`. Since it is always hardcoded to one million, I don't think checked div would make any difference.
However, checked operations might make sense before multiplication and addition, as there should be no overflows there.